src-built-in/components/toolbarRefactored/toolbarHotkeys.ts

```diff
@@ -0,0 +1,109 @@
+const keys = FSBL.Clients.HotkeyClient.keyMap;
+
+/**
+ * Helper so I didn't have to rewrite the same 3 lines of code 6
+ * times.
+ * @param err Unlikely error; would be coming in when our hotkey is registered.
+ */
+const handleHotkeyRegistrationError = (err:any) => {
+	if (err) {
+		FSBL.Clients.Logger.error(`HotkeyClient.addGlobalHotkey failed, error:`, err);
+	}
+}
+/**
+	 * Function to bring toolbar to front (since dockable toolbar can be hidden)
+	 * The search input box will be open and any previous results will be displayed
+	 * @param {boolean} focus If true, will also focus the toolbar
+	 * @TODO Evaluate whether focus is necessary or if it's just vestigial.
+	 */
+const bringToolbarToFront = (focus = false) => {
+	finsembleWindow.bringToFront(null, (err:any) => {
+		if (err) {
+			FSBL.Clients.Logger.error(`bringToolbarToFront failed, error:`, err);
+		}
+
+		if (focus) {
+			finsembleWindow.focus();
+			// Leaving this line of code because clearly someone ran into something difficult.
+			// self.Store.setValue({ field: "searchActive", value: false });
+		}
+	});
+}
+
+ 	/**
+* Hides the toolbar
+*/
+const hideToolbar = () => {
+ finsembleWindow.blur();
+ finsembleWindow.hide();
+}
+
+/**
+ * Shows the toolbar left-aligned to the user's mouse.
+ */
+const showToolbar = () => {
+	FSBL.Clients.WindowClient.showAtMousePosition();
+	bringToolbarToFront(true);
+}
+
+/**
+ * Mimimizes all windows.
+ */
+const minimizeAllWindows = () => {
+	FSBL.Clients.WorkspaceClient.minimizeAll();
+}
+
+/**
+ * Brings all windows to front.
+ */
+const revealAllWindows = () => {
+	FSBL.Clients.LauncherClient.bringWindowsToFront()
+}
+
+const registerShowToolbarHotkey = () => {
+	FSBL.Clients.HotkeyClient.addGlobalHotkey(
+		[keys.ctrl, keys.alt, keys.t],
+		(err: any) => {
+			handleHotkeyRegistrationError(err);
+			showToolbar();
+		}
+	);
+}
+
+const registerHideToolbarHotkey = () => {
+	FSBL.Clients.HotkeyClient.addGlobalHotkey(
+		[keys.ctrl, keys.alt, keys.h],
+		(err: any) => {
+			handleHotkeyRegistrationError(err);
+			hideToolbar();
+	});
+}
+
+const registerRevealAllHotkey = () => {
+	FSBL.Clients.HotkeyClient.addGlobalHotkey([keys.ctrl, keys.alt, keys.up], (err: any) => {
+		handleHotkeyRegistrationError(err);
+		revealAllWindows();
+	});
+}
+
+const registerMinimizeAllHotkey = () => {
+	FSBL.Clients.HotkeyClient.addGlobalHotkey([keys.ctrl, keys.alt, keys.down], (err: any) => {
+		handleHotkeyRegistrationError(err);
+		minimizeAllWindows();
+	});
+}
+
+const registerShowSearchHotkey = () => {
+	FSBL.Clients.HotkeyClient.addGlobalHotkey([keys.ctrl, keys.alt, keys.f], (err: any) => {
+		handleHotkeyRegistrationError(err);
+		bringToolbarToFront(true);
+	});
+}
+
```

Comment:
Are we keeping the hot key for favorites items? If so we probably need to set up the favorite hotkeys here as well.

Reply:
So, they are not in the requirements, so I didn't add them. They're a big piece of complication. I can ask terry and Sean if you'd like.

Reply:
Sure! In case we missed a functionality when writing the requirement

Reply:
Okay, discussing with them. I think it's a super undiscoverable feature and should be removed in either case. I didn't know anyone outside of me and nicolai knew it existed.